datapath: Filter out bpftool probes emitting dmesg messages

vadorovsky opened the pull request:
bpftool feature probes related to trace, perf and write_user helpers are
emitting dmesg messages with warnings which may be confusing for
operators running Cilium on production environments. After this change,
those probes will be not performed.

Fixes #10048

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

```release-note
Filter out bpftool probes emitting dmesg messages
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/10164)
<!-- Reviewable:end -->

vadorovsky committed:
datapath: Filter out bpftool probes emitting dmesg messages

bpftool feature probes related to trace, perf and write_user helpers are
emitting dmesg messages with warnings which may be confusing for
operators running Cilium on production environments. After this change,
those probes will be not performed.

Fixes #10048

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

vadorovsky committed:
Dockerfile: Use Cilium fork of the kernel to build bpftool

Cilium fork of the Linux kernel contains necessary enhancements for
bpftool which are not avalavle upstream yet.

Ref: #10048

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

qmonnet commented on pkg/datapath/linux/probes/probes.go:
```diff
@@ -133,8 +133,10 @@ func NewProbeManager() *ProbeManager {
-			log, true)
+			defaults.ExecTimeout, "bpftool", "-j", "feature",
+			"probe", "filter_out",
+			"'\\(trace\\|perf\\|write_user\\)'",
```
Are perf-related helpers printing kernel logs as well?
Filtering out on `perf` will also remove probe results for `BPF_MAP_TYPE_PERF_EVENT_ARRAY` and related helpers, is that desirable?

vadorovsky replied:
Hmm... I checked and filtering out `perf` doesn't seem to be necessary.

So `bpftool -j feature probe filter_out '\(trace\|write_user\)'` works. I will change that. Thanks!

vadorovsky replied:
done

qmonnet reviewed (approved):
Pending the merge of the Linux PR, we're good to go with this one. Thanks!

vadorovsky committed:
ci: Install bpftool from Cilium fork of the kernel

This should be done in packer-ci-build, but to get the fix faster, we
install patched bpftool here as a temporary hack...

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>
